internal/dev_server/sdk/go_sdk_test.go

```diff
@@ -123,4 +124,43 @@ func TestSDKRoutesViaGoSDK(t *testing.T) {
 			assert.Equal(t, value.AsArbitraryValue(), flagUpdate.NewValue.AsArbitraryValue())
 		})
 	}
+
+	newUpdates := map[string]ldvalue.Value{
+		"boolFlag":   ldvalue.Bool(true),
+		"stringFlag": ldvalue.String("fool"),
+		"intFlag":    ldvalue.Int(567),
+		"doubleFlag": ldvalue.Float64(99.99),
+		"jsonFlag":   ldvalue.CopyArbitraryValue(map[string]any{"tortoise": "hare"}),
+	}
+
+	// This test is testing the "put" payload in a roundabout way by verifying each of the flags are in there.
+	t.Run("Sync sends full flag payload for project", func(t *testing.T) {
+		trackers := make(map[string]*<-chan interfaces.FlagValueChangeEvent, len(updates))
+
+		for flagKey, newValue := range newUpdates {
+			flagUpdateChan := ld.GetFlagTracker().AddFlagValueChangeListener(flagKey, ldContext, ldvalue.String("uh-oh"))
+			defer ld.GetFlagTracker().RemoveFlagValueChangeListener(flagUpdateChan)
+			trackers[flagKey] = &flagUpdateChan
+
+			_, err := store.UpsertOverride(ctx, model.Override{
```

Comment:
This is probably OK, but it is kinda a hack since we're calling the store directly in what's supposed to be more of an integration test (although not quite since we're mocking out the external stuff anyways).

WDYT about a more realistic scenario where we test resync before we test overrides and we change the mock response from the SDK to have different values and higher version numbers?

Reply:
I'm down for that